### ISSUE TITLE
eSIM Multiple Enabled Profiles

### ISSUE THREAD

**Issue by Majesty6338** [GrapheneOS/os-issue-tracker#2098]
As I know Android can now support dual eSIM (MEP). 
https://source.android.com/docs/core/connect/esim-mep
But on the latest version of GrapheneOS and on Pixel 7 this functionality does not work and gives an error: "Only one downloaded SIM can be active at a time".

```
$ adb shell pm list features | grep android.hardware.telephony.euicc.mep
feature:android.hardware.telephony.euicc.mep
```
```
$ adb logcat
04-10 09:49:09.349 10949 10949 I ToggleSubscriptionDialogActivity: isMultipleEnabledProfilesSupported():false
04-10 09:49:09.349 10949 10949 D ToggleSubscriptionDialogActivity: Handle subscription enabling.
04-10 09:49:09.349 10949 10949 D ToggleSubscriptionDialogActivity: DSDS is already enabled. Condition not satisfied.
04-10 09:49:09.373 10949 10949 I ConfirmDialogFragment: Showing dialog with title =Switch to [***]?
```
Two of my providers say they support eSim MEP.
What could be wrong?

-----
```
[ro.build.fingerprint]: [google/panther/panther:13/TQ2A.230305.008/2023040400:user/release-keys]
```

**Comment by FullStackSoftwareLLC** [comment]
following...
I'm having the same problem with android 13 on a pixel 5a utilizing AT&T and Vodafone

**Comment by Shapes0** [comment]
Same issue with AT&T and T-Mobile for me on my Pixel 7

**Comment by erroriel** [comment]
I'm unable to properly activate service on Helium Mobile on a Pixel 7 without this feature on GrapheneOS due to them relying on multiple eSIMs being active simultaneously.

https://support.google.com/pixelphone/answer/9449293?hl=en
"On Pixel 7 and Pixel 7 Pro phones and if your carrier allows, you can use 2 eSim profiles simultaneously."

https://support.hellohelium.com/en/articles/7130896-installing-and-activating-your-sims
"Voice eSIM (Google Pixel 7 and 7 Pro)" and "Data eSIM (Google Pixel models)"

https://support.hellohelium.com/en/articles/7240207-supported-devices
"Users of devices with Dual eSIM compatibility should continue to use a Voice and Data eSIM."

**Comment by simpsonstuart** [comment]
Has there been any updates for this? I am facing the same issue on helium mobile where I have to use dual esim. I tested the stock os with my 7pro and both esims at once work just fine, so I believe this is just a bug where graphene isn't getting if the phone supports MEP properly for supported hardware.

**Comment by ryanriske** [comment]
I also have a need for dual active esims on a Pixel 7 Pro and would appreciate it if this issue was addressed.

**Comment by FullStackSoftwareLLC** [comment]
> I also have a need for dual active esims on a Pixel 7 Pro and would appreciate it if this issue was addressed.

https://www.androidpolice.com/dual-esim-pixel-7/

**Comment by ryanriske** [comment]
> > I also have a need for dual active esims on a Pixel 7 Pro and would appreciate it if this issue was addressed.
> 
> https://www.androidpolice.com/dual-esim-pixel-7/

Yes, the phone supports it with the stock operating system but GrapheneOS does not yet. It disables the active esim when I try to activate a second one stating "Only one downloaded SIM can be active at a time."

**Comment by Shapes0** [comment]
On Android 14 now and still no dual eSIM

**Comment by jnelsontn** [comment]
> On Android 14 now and still no dual eSIM

Has this been fixed?

**Comment by Shapes0** [comment]
> > On Android 14 now and still no dual eSIM
> 
> Has this been fixed?

Sadly no, I still can't enable multiple eSIMs on the latest update.

**Comment by Shapes0** [comment]
@thestinger is it possible to get insight on the extent that GOS can support/port this function present in the stock OS?

**Comment by muhomorr** [comment]
I'm unable to reproduce this issue on Pixel 7 and on Pixel 8: activated 3 eSIMs, 2 eSIMs are enabled at the same time.

Check again whether this is still an issue on the latest OS release.

**Comment by erroriel** [comment]
Still not working for me on a Pixel 7 with the latest release.

![Screenshot_20240128-131420.png](https://github.com/GrapheneOS/os-issue-tracker/assets/56495874/0ef19dd3-f29e-4bfc-b0ea-c8b5beff677c)

![Screenshot_20240128-131458.png](https://github.com/GrapheneOS/os-issue-tracker/assets/56495874/a86216b8-33d2-403a-b893-f574b942098e)

**Comment by Shapes0** [comment]
Also not working for me on 20240126 on my Pixel 7. I have tried to disable and re-enable eSIM support to no avail. I wonder if a wipe and reinstall is necessary to get it working.

**Comment by muhomorr** [comment]
@erroriel @Shapes0 Post your carrier settings versions (Settings -> Network -> SIMs -> <SIM> -> Long press "Carrier settings version" -> Copy)

**Comment by Shapes0** [comment]
> @erroriel @Shapes0 Post your carrier settings versions (Settings -> Network -> SIMs -> -> Long press "Carrier settings version" -> Copy)

tmobile_us-52000000043
MCC: 310 MNC: 240
2023-09-27

**Comment by erroriel** [comment]
Physical SIM:
```
att5g_us-52000000018
MCC: 310 MNC: 410
MVNO: gid: 53FF
2023-08-24
```

eSIM 1:
```
tmobile_us-52000000043
MCC: 310 MNC: 260
2023-09-27
```

eSIM 2 and 3:
```
tmobile_us-52000000043
MCC: 310 MNC: 240
2023-09-27
```

**Comment by muhomorr** [comment]
@Shapes0 @erroriel T-Mobile US doesn't allow to use eSIM MEP:

> Your Google Pixel 7 lets you store multiple eSIM profiles, but only one can be used at a time

https://www.t-mobile.com/support/tutorials/device/google/pixel-7/topic/esim/how-to-switch-between-esim-profiles

I've found several reports online that state that MEP doesn't work on T-Mobile US.

**Comment by erroriel** [comment]
@muhomorr Please see above about Helium. This is a supported setup that only does not work on GrapheneOS. I have tested this successfully with stock and other custom ROMs.

**Comment by muhomorr** [comment]
@erroriel Have you tested it on the same device?

Check whether DSDS (dual SIM, dual standby) is enabled in phone testing menu:
dial `*#*#4636#*#*`, tap Phone information, then scroll down to "Enable DSDS".

**Comment by erroriel** [comment]
No, not on this device specifically, but on another testing/development Pixel 7 because of having to factory reset to change ROMs.

Before I originally joined this issue, I downloaded two cheap prepaid Tello (T-Mobile MVNO) eSIMs on my test device and successfully activated them both simultaneously on stock and another custom ROM. Then I installed GrapheneOS and could not activate them simultaneously, only swap. I kept the eSIMs on the device for each ROM tested, meaning I did not redownload them each time. I can repeat this experiment on the current build if that would help narrow things down.

On this device, DSDS is enabled and I am using it with the physical SIM and one eSIM. No difference to MEP after toggling off and on.

What carriers are you testing with? @muhomorr

**Comment by muhomorr** [comment]
@erroriel 

> I downloaded two cheap prepaid Tello (T-Mobile MVNO) eSIMs on my test device and successfully activated them both simultaneously on stock and another custom ROM

Did you have a physical SIM active at the time? If there's a physical SIM active, at most one eSIM can be enabled.

> What carriers are you testing with?

LTE carrier from Eastern Europe.

**Comment by muhomorr** [comment]
@erroriel Check the log for `isMultipleEnabledProfilesSupported` text after trying to enable the second eSIM. Can do it without adb access, by going to Settings -> System -> View logs and searching for that text.

I get `ToggleSubscriptionDialogActivity: isMultipleEnabledProfilesSupported():true` on both Pixel 7 and 8 (unlike the OP).

**Comment by Shapes0** [comment]
![Screenshot_20240131-134449~2_20240131134555](https://github.com/GrapheneOS/os-issue-tracker/assets/63262012/421a140a-b40d-498d-8c3c-d3c50b7954d4)
It looks to be false on my device

**Comment by Shapes0** [comment]
As an update to this, I reinstalled GrapheneOS on my phone and now I have the ability to enable both eSIMs. It seems that updates never triggered some change made in the fresh install image.

**Comment by crx60** [comment]
I'm facing the same issue.

> Resetting cellular configuration might have worked. I don't think carrier configurations that are in use are updated without provisioning again.

Assuming you're referring to _settings_ > _reset options_ > _reset mobile network settings_ (_erase esims_ checked), that didn't help. I'm also get `ToggleSubscriptionDialogActivity: isMultipleEnabledProfilesSupported():false` in the logs.

**Comment by kouyk** [comment]
Has the issue been fixed? I tried to enable a second esim on the September build on a Pixel 7 and it is still not working.

**Comment by kouyk** [comment]
> In Settings > System > Reset, you can reset cellular settings. Try that.

I've tried that it didn't help, though I can't confirm that my carrier supports this or not. Or perhaps I will need a fresh install that is not quite feasible.

**Comment by xesxen** [comment]
Same for me. I have one KPN (NL) e-sim and one KPN MVNO e-sim. Resetting the cellular settings doesn't allow me to enable both. DSDS is enabled according to *#*#4636#*#*.

**Comment by xesxen** [comment]
@thestinger Pixel 7, GVU6C. Currently running build 2024090400.

**Comment by kouyk** [comment]
> @kouyk Which carrier(s)?

I am in SG, I have Eight (MVNO on StarHub), Eskimo (MVNO based on Singtel AFAIK).

**Comment by ryanriske** [comment]
I'm also still experiencing this problem on a Pixel 7 Pro GE2AE (purchased unlocked from Google directly) running 2024090400.

I currently have a Verizon MVNO eSIM but have also run a T-Mobile US first party physical SIM, T-Mobile MVNO eSIM, and an international eSIM from 1GLOBAL.

Resetting the cellular settings did not enable the ability to use two eSIMs at once no matter which service's SIM was installed and/or active.

**Comment by laf0rge** [comment]
@no-usernames-left yes, that is correct. There are a number of high-end smartphones that support v3.x and MEP though.  It's sufficient for 1-2 large eUICC / CardOS vendors to have supporting products to make it wide-spread in the top segment of the market.

Quantity of devices with v3.x != number of CardOS/eUICC vendors with v3.x.

**Comment by ryanriske** [comment]
> Resetting the cellular settings did not enable the ability to use two eSIMs at once no matter which service's SIM was installed or active. I have not tried flashing back to the stock OS and reinstalling GrapheneOS yet.

I flashed the latest current stock OS, enabled two eSIMs successfully, and flashed back GrapheneOS again. That apparently set the flag that allows support for using two eSIMs simultaneously, and I can successfully toggle between pSIM+eSIM and eSIM+eSIM from within GrapheneOS now.

As an aside, the built-in backup and restore functionality works very well and made that process a lot less painful than I was expecting.

**Comment by no-usernames-left** [comment]
@ryanriske Which model of device are you using?

Did you try doing so from within GrapheneOS first? If so, what was the result?

>I can successfully toggle between pSIM+eSIM and eSIM+eSIM from within GrapheneOS now.

Where exactly is that toggle located?

**Comment by thestinger** [comment]
You should be able to just toggle on DSDS in GrapheneOS but it should get toggled on automatically. Check earlier in the thread for DSDS.

**Comment by no-usernames-left** [comment]
I don't think they are referring to the DSDS toggle.

**Comment by ryanriske** [comment]
> [@ryanriske](https://github.com/ryanriske) Which model of device are you using?
>
> Did you try doing so from within GrapheneOS first? If so, what was the result?
 
It's a Pixel 7 Pro GE2AE purchased unlocked from Google directly. Yes, I've tried multiple times over the years this issue has been open but it always failed with the error in the OP. I flashed my phone to GrapheneOS before Google pushed the dual eSIM feature out in a Pixel Drop update, so perhaps the code to toggle that flag is not part of AOSP.

> > I can successfully toggle between pSIM+eSIM and eSIM+eSIM from within GrapheneOS now.
> 
> Where exactly is that toggle located?

Settings -> Network & internet -> SIMs

Just try to enable one eSIM for calls/SMS and another eSIM for data as you normally would.

**Comment by no-usernames-left** [comment]
> Settings -> Network & internet -> SIMs

What is the name of the toggle to which you were referring?

>I can successfully toggle between pSIM+eSIM and eSIM+eSIM from within GrapheneOS now.

**Comment by ryanriske** [comment]
> What is the name of the toggle to which you were referring?

I meant just try to enable two eSIMs at the same time as you normally would. There's no specific toggle in the UI that enables dual eSIM.

**Comment by Jabb0** [comment]
> Does anyone have issues with this on 8th/9th generation Pixels? Is everyone with issues on 7th generation?

@thestinger Yes I do. I have a Pixel 7 Pro on the latest release graphene OS.
When I activate one eSIM the other one gets disabled.
